### ISSUE TITLE
Where is the code for to perturb B?

### ISSUE THREAD

**Issue by bongbonglemon** [senguptaumd/Background-Matting#7]
Where is the code to get the "perturbed version" of B? Or are the background images in the sample data already the perturbed versions?

**Comment by senguptaumd** [comment]
Can you be a bit more specific about the "perturb version" of B? Sorry, I can not remember where I used this term, and for what context.
The background images in sample data are raw backgrounds. You have to align it with our data based on the provided code.

**Comment by gruossomonica** [comment]
I think @bongbonglemon refers to _B'_, which is defined in the paper as the perturbed version of the true background _B_. Correct me if I'm wrong.
I'm interested in your work, so sometimes I take a look at your GitHub page. :)

**Comment by senguptaumd** [comment]
We can write the captured image I=\alpha*F + (1-\alpha)*B, where B is the true background. Now we capture background when the subject leaves the scene. This background is not the true background. Why is it not the true background? Because there are small changes w.r.t. shadows, reflections, camera resampling etc. Thus we call this captured background as B'.

We train on synthetic-composite Adobe dataset, which provides us with image, alpha matte, and foreground (I, F,\alpha). We can easily calculate the true background B. But we do not know how the captured background B would look like. Thus we perturb B to obtain B' by adding sample gaussian noise, gamma correction etc.

**Comment by bongbonglemon** [comment]
@senguptaumd Yes I understand why you did it. I was asking as to where the code was for the  "adding sample gaussian noise, gamma correction etc."

**Comment by senguptaumd** [comment]
Sorry, I do not have the training code ready yet. It will take some time. But you can follow what is mentioned in the paper. I do plan to release the training code on synthetic and real data shortly.

**Comment by bongbonglemon** [comment]
Alright. Yeah I saw that you mentioned that in the README but I was half guessing that it should be in the preprocessing script. And yes I would like to implement it myself. I'm just a bit confused by what you mean by adding the gaussian noice and gamma correction "around the foreground region". Does that refer to the area that subject was covering? And does "around" refer to the just the perimeter or is it the whole area? Thank you for time!

**Comment by senguptaumd** [comment]
Pre-processing script for testing only does perspective alignment and bias-gain correction. Perturbing B' is only needed for training.

Given the foreground segmentation mask, m, dilate the mask up to certain steps, say cv2.dilate() with 100 steps to create a new mask m1. Add gaussian noise in the (m1-m) region. It may not be super important, as long as you add some Gaussian noise with random mu, sigma and some gamma change it is fine.
